packages/block-editor/src/hooks/color.js

```diff
@@ -303,17 +303,15 @@ export function ColorEdit( props ) {
 
 	const onChangeLinkColor = ( value ) => {
 		const colorObject = getColorObjectByColorValue( colors, value );
-		props.setAttributes( {
-			style: {
-				...props.attributes.style,
-				color: {
-					...props.attributes.style?.color,
-					link: colorObject?.slug
-						? `var:preset|color|${ colorObject.slug }`
-						: value,
-				},
-			},
-		} );
+		const newStyle = setWith(
+			style ? clone( style ) : {},
+			[ 'elements', 'link', 'color', 'text' ],
+			colorObject?.slug
+				? `var:preset|color|${ colorObject.slug }`
+				: value,
+			clone
```

Comment:
I'm confused about how this works and [the docs](https://lodash.com/docs/4.17.15#setWith) aren't of much help. Is there an alternative way to rewrite this? Wouldn't this work (pseudo-code)?

```js
props.setAttributes(
  style: {
    ...existingStyle,
   elements.link.color.text: <preset-or-raw-value>
  }
)
```